factcast-store-pgsql/src/main/java/org/factcast/store/pgsql/validation/http/IndexFetcher.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright © 2017-2020 factcast.org
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.factcast.store.pgsql.validation.http;
+
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.Optional;
+
+import org.factcast.store.pgsql.validation.schema.SchemaRegistryUnavailableException;
+import org.springframework.beans.factory.config.ConfigurableBeanFactory;
+import org.springframework.context.annotation.Scope;
+import org.springframework.stereotype.Component;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import okhttp3.Request;
+import okhttp3.Request.Builder;
+import okhttp3.Response;
+
+/**
+ * fetches the index (using if-modified-since)
+ *
+ * @author uwe
+ *
+ */
+@Slf4j
+@Component
+@Scope(scopeName = ConfigurableBeanFactory.SCOPE_SINGLETON)
+@RequiredArgsConstructor
+public class IndexFetcher {
+
+    private final URL schemaRegistryUrl;
+
+    private String since;
+
+    private String etag;
+
+    /**
+     * @return future for empty, if index is unchanged, otherwise the updated
+     *         index.
+     */
+    Optional<RegistryIndex> fetchIndex() {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use explicit scoping instead of the default package private level](https://app.codacy.com/manual/uwe/factcast/pullRequest?prid=4877539)